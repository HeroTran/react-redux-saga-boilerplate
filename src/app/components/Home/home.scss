.wrapper-home{
    display: block;
    width: 100%;
    position: relative;
    h1{
        font-size: 30px;
        color: red;
        display: inline-block;
    }
    p{
        font-size: 30px;
        color: blue;
        display: inline-block;
    }
}