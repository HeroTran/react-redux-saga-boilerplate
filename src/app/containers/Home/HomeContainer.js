import { connect } from 'react-redux';
import Home from '../../components/Home/Home';

const mapStateToProps = (state) => {
  return {};
};

const mapDispatchToProps = (dispatch) => {
  return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
