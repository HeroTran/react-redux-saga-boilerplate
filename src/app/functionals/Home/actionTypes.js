export const actionTypes = {
  GET_HOME_REQUEST: 'GET_HOME_REQUEST',
  GET_HOME_SUCCESS: 'GET_HOME_SUCCESS',
  GET_HOME_FAILURE: 'GET_HOME_FAILURE',
};
export default actionTypes;
