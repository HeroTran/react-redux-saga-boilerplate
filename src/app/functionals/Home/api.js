import * as apiHelper from '../../utils/apiHelper';
import { API_URL } from './constants';

export function getHome() {
  const urlParam = ``;
  const request = {
    url: `${API_URL.API_HOME}${urlParam}`,
  };
  return apiHelper.apiGet(request);
}
