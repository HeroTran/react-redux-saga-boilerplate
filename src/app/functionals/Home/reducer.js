import { fromJS } from 'immutable';
import { PAGE_FORMAT } from '../Common/constants';

const initialState = fromJS({
  home: [],
  pagination: {
    current_page: PAGE_FORMAT.PAGE_DEFAULT,
    page_size: PAGE_FORMAT.SIZE_FIVE,
    total: PAGE_FORMAT.TOTAL_DEFAULT,
  },
  error: null,
});

export default function homeReducer(state = initialState, action) {
  switch (action.type) {
    default:
      return state;
  }
}
