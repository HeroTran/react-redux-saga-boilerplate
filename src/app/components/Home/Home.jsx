import React from 'react';
import i18n from '../../i18n';
import './home.scss';

const Home = (props) => {
  return (
    <div className="wrapper-home">
      <h1>Welcome you !</h1>
      <p>{i18n.t('common.about')}</p>
    </div>
  );
};

export default Home;
