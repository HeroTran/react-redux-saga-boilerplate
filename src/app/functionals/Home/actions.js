import actionTypes from './actionTypes';

export function getHometRequest(params) {
  return {
    type: actionTypes.GET_HOME_REQUEST,
    params,
  };
}

export function getHometSuccess(data) {
  return {
    type: actionTypes.GET_HOME_SUCCESS,
    data,
  };
}

export function getHometFailure(error) {
  return {
    type: actionTypes.GET_HOME_FAILURE,
    error,
  };
}
