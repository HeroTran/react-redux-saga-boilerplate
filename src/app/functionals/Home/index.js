import * as homeAction from './actions';
import homeReducer from './reducer';
import homeSaga from './saga';
import * as homeSelectors from './selectors';

export { homeAction, homeReducer, homeSaga, homeSelectors };
