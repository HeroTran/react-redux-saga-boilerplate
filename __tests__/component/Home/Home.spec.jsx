/*
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Home from '../../../app/renderer/components/Home/Home';

let wrapper;
let getClassList;
beforeEach(() => {
  getClassList = jest.fn();
  const classList = [
    {
      code: 'B00744',
      booking_type: 1,
      room_id: 'b6991a15-f9ea-4746-af0c-d637dd66ee8e',
      time: '2020-05-31 06:00',
      availability_type: 1,
      availability_status: 'BOOKED',
      user_id: 2616,
      teacher_name: 'Tran Van Hung',
      learner_code: 'L00333',
      learner_name: 'Hero Tran',
      learner_next_level: '02051002',
      parent_code: 'P03140',
      lesson_code: null,
      feedbacks: [],
      reason: null,
      status: 'INIT',
    },
  ];
  const pagination = {
    current_page: 1,
    page_size: 5,
    total: 0,
  };
  wrapper = (
    <MemoryRouter>
      <Home
        classList={[]}
        pagination={pagination}
        getClassList={getClassList}
      />
    </MemoryRouter>
  );
});
describe('should be rendered without error', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
*/