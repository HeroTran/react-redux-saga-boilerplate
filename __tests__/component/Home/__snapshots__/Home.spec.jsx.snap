// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be rendered without error matches snapshot 1`] = `
<div
  className="wrapper-home"
>
  <img
    alt="background-home"
    className="each-svg"
    src="test-file-stub"
  />
  <img
    alt="background-home"
    className="each-bg"
    src="test-file-stub"
  />
</div>
`;
